cnd/src/swap_protocols/metadata_store.rs

```diff
@@ -85,49 +82,86 @@ impl Metadata {
     }
 }
 
-#[derive(Debug, Fail)]
+#[derive(Debug)]
 pub enum Error {
-    #[fail(display = "Metadata already exists")]
-    DuplicateKey,
+    Path(String),
+    Init(String),
+    Connect(String),
+    Load(String),
+    Insert(String),
+    Parse(String),
+}
+
+impl fmt::Display for Error {
```

Comment:
Usually, there is no need for a `Display`-impl. Why do you need it?

Reply:
I got tangled working with converting from one type to another using `ToSt`, `FromStr`, `From`, and constructors.  This may be a hang over from that - I'll investigate.

Reply:
> Does it though? For me, aliases cause major headache usually because you cannot find the type when you are looking for it ..

Are you referring to the `pub use ...`?  if so it seems to not need the pub.  Will remove that.

Reply:
> When are we checking for 1 here?
> I would expect this call to return you the created primary key for the row, hence this is only going to run once for every user (there is only one swap with ID 1).

We don't need to return the primary key because we do not create it, its the swap id.